man/partition.d.xml

```diff
@@ -0,0 +1,385 @@
+                <entry>Root file system partition for the x86 (32bit, aka i386) architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-x86-verity</constant></entry>
+                <entry>Verity data for the x86 (32bit) root file system partition</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-x86-64</constant></entry>
+                <entry>Root file system partition for the x86_64 (64bit, aka amd64) architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-x86-64-verity</constant></entry>
+                <entry>Verity data for the x86_64 (64bit) root file system partition</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-arm</constant></entry>
+                <entry>Root file system partition for the ARM (32bit) architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-arm-verity</constant></entry>
+                <entry>Verity data for the ARM (32bit) root file system partition</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-arm64</constant></entry>
+                <entry>Root file system partition for the ARM (64bit, aka aarch64) architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-arm64-verity</constant></entry>
+                <entry>Verity data for the ARM (64bit, aka aarch64) root file system partition</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-ia64</constant></entry>
+                <entry>Root file system partition for the ia64 architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-ia64-verity</constant></entry>
+                <entry>Verity data for the ia64 root file system partition</entry>
+              </row>
+            </tbody>
+          </tgroup>
+        </table>
+
+        <para>This setting defaults to <constant>linux-generic</constant>.</para>
+
+        <para>Most of the partition type UUIDs listed above are defined in the <ulink
+        url="https://systemd.io/DISCOVERABLE_PARTITIONS">Discoverable Partitions
+        Specification</ulink>.</para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>Label=</varname></term>
+
+        <listitem><para>The textual label to assign to the partition if none is assigned yet. Note that this
+        setting is not used for matching. It is also not used when a label is already set for an existing
+        partition. It is thus only used when a partition is newly created or when an existing one had a no
+        label set (that is: an empty label). If not specified a label derived from the partition type is
+        automatically used.</para></listitem>
+      </varlistentry>
```

Comment:
The big difference is that network files match existing objects (i.e. net devices), and then apply configuration to it, they do not create objects and there's an n:1 relationship between these devices and the configuration files.

This is inherently different here with systemd-repart. here we define objects that shall exist, and there's a 1:1 relationship between our configuration files and the partitions on disk. i.e. it's more about describing stuff *to create*, instead of *matching* what already exists, and we never want to apply a single configuration to multiple objects.

or to say this differently i think the partition files are better compared with *.netdev than with *.network. The netdev files describe stuff to create, and they match existing stuff only by ifname, which is in the [NetDev] section, and not the [Match] section.

Does that make sense?

Or to extend on your example above: presumably by your configuration you#d expect that the partition type set when the snippet is applied would be initialized as the esp type uuid, right? so you'd populate it from a setting of the [Match] section? but that's pretty skewed, no? So you'd then have another Type= setting in the [Partition] to fix that skewedness, but it would be redundant in almost all cases, since the Type= in the [Match] and in the [Partition] section would always be set to the same value. So I ultimately then decided: let's just drop the [Match] section, and match only by Type.

Reply:
OK, you make a good point. Though I think we might want to allow matching by partition label in the future.